polkit/authority.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package polkit
+
+import (
+	"github.com/godbus/dbus"
+)
+
+type CheckAuthorizationFlags uint32
+
+const (
+	CheckAuthorizationNone                 CheckAuthorizationFlags = 0x00
+	CheckAuthorizationAllowUserInteraction CheckAuthorizationFlags = 0x01
+)
+
+// CheckAuthorization queries polkit to determine whether a subject is
+// authorized to perform an action.
```

Comment:
I used the D-Bus API and existing C binding as a guide for the interface.  I guess it depends on whether we want something that only covers our immediate use case, or something that can be reused.

The result struct is described here:

https://www.freedesktop.org/software/polkit/docs/latest/eggdbus-interface-org.freedesktop.PolicyKit1.Authority.html#eggdbus-struct-AuthorizationResult

The main pieces of additional information are:

1. if the user is authorised successfully, and policy allows that authorisation to be retained for a time, it provides an opaque ID that can be used to cancel the retained authorisation.
2. if you ask for non-interactive auth, it can tell you that authorisation could succeed with interaction.
3. differentiate between "access denied" and "user cancelled the auth dialog".

(2) and (3) look like they could be handled via custom errors.  I'm not sure how important (1) is.